File: recovered/src/com/lumiyaviewer/lumiya/ui/chat/profiles/UserAboutTextEditFragment.java
package com.lumiyaviewer.lumiya.ui.chat.profiles;

import android.content.Context;
import android.os.Bundle;
import com.lumiyaviewer.lumiya.R;
import com.lumiyaviewer.lumiya.slproto.SLAgentCircuit;
import com.lumiyaviewer.lumiya.slproto.SLMessage;
import com.lumiyaviewer.lumiya.slproto.messages.AvatarPropertiesReply;
import com.lumiyaviewer.lumiya.slproto.users.ChatterID;
import com.lumiyaviewer.lumiya.ui.common.ChatterFragment;

/* loaded from: classes.dex */
public class UserAboutTextEditFragment extends ProfileTextFieldEditFragment {
    private static final String IS_FIRST_LIFE_KEY = "isFirstLife";
    private AvatarPropertiesReply avatarProperties;

    private boolean isFirstLife() {
        Bundle arguments = getArguments();
        if (arguments != null) {
            return arguments.getBoolean(IS_FIRST_LIFE_KEY);
        }
        return false;
    }

    public static Bundle makeSelection(ChatterID chatterID, boolean z) {
        Bundle makeSelection = ChatterFragment.makeSelection(chatterID);
        makeSelection.putBoolean(IS_FIRST_LIFE_KEY, z);
        return makeSelection;
    }

    @Override // com.lumiyaviewer.lumiya.ui.common.ChatterFragment
    protected String decorateFragmentTitle(String str) {
        return getString(R.string.edit_about_title, str);
    }

    @Override // com.lumiyaviewer.lumiya.ui.common.TextFieldEditFragment
    protected String getFieldHint(Context context) {
        return getString(R.string.edit_about_hint);
    }

    @Override // com.lumiyaviewer.lumiya.ui.chat.profiles.ProfileTextFieldEditFragment
    /* renamed from: onAvatarProperties */
    protected void m517x7aa22308(AvatarPropertiesReply avatarPropertiesReply) {
        this.avatarProperties = avatarPropertiesReply;
        setOriginalText(isFirstLife() ? SLMessage.stringFromVariableOEM(this.avatarProperties.PropertiesData_Field.FLAboutText) : SLMessage.stringFromVariableUTF(avatarPropertiesReply.PropertiesData_Field.AboutText));
    }

    @Override // com.lumiyaviewer.lumiya.ui.common.TextFieldEditFragment
    protected void saveEditedText(SLAgentCircuit sLAgentCircuit, ChatterID chatterID, String str) {
        if (this.avatarProperties != null) {
            String stringFromVariableUTF = SLMessage.stringFromVariableUTF(this.avatarProperties.PropertiesData_Field.AboutText);
            String stringFromVariableOEM = SLMessage.stringFromVariableOEM(this.avatarProperties.PropertiesData_Field.FLAboutText);
            if (isFirstLife()) {
                stringFromVariableOEM = str;
            } else {
                stringFromVariableUTF = str;
            }
            sLAgentCircuit.getModules().userProfiles.UpdateAvatarProperties(this.avatarProperties.PropertiesData_Field.ImageID, this.avatarProperties.PropertiesData_Field.FLImageID, stringFromVariableUTF, stringFromVariableOEM, (this.avatarProperties.PropertiesData_Field.Flags & 1) != 0, (this.avatarProperties.PropertiesData_Field.Flags & 2) != 0, SLMessage.stringFromVariableOEM(this.avatarProperties.PropertiesData_Field.ProfileURL));
        }
    }
}

File: app/src/main/java/com/lumiyaviewer/lumiya/ui/chat/profiles/UserAboutTextEditFragment.kt
package com.lumiyaviewer.lumiya.ui.chat.profiles

import android.content.Context
import android.os.Bundle
import com.lumiyaviewer.lumiya.R
import com.lumiyaviewer.lumiya.slproto.SLAgentCircuit
import com.lumiyaviewer.lumiya.slproto.SLMessage
import com.lumiyaviewer.lumiya.slproto.messages.AvatarPropertiesReply
import com.lumiyaviewer.lumiya.slproto.users.ChatterID
import com.lumiyaviewer.lumiya.ui.common.ChatterFragment

class UserAboutTextEditFragment : ProfileTextFieldEditFragment() {
    private var avatarProperties: AvatarPropertiesReply? = null

    private fun isFirstLife(): Boolean = arguments?.getBoolean(IS_FIRST_LIFE_KEY) ?: false

    override fun decorateFragmentTitle(title: String): String = getString(R.string.edit_about_title, title)

    override fun getFieldHint(context: Context): String = getString(R.string.edit_about_hint)

    override fun onAvatarProperties(avatarPropertiesReply: AvatarPropertiesReply) {
        avatarProperties = avatarPropertiesReply
        setOriginalText(
            if (isFirstLife()) {
                SLMessage.stringFromVariableOEM(avatarPropertiesReply.PropertiesData_Field.FLAboutText)
            } else {
                SLMessage.stringFromVariableUTF(avatarPropertiesReply.PropertiesData_Field.AboutText)
            }
        )
    }

    override fun saveEditedText(agentCircuit: SLAgentCircuit, chatterID: ChatterID, text: String) {
        val properties = avatarProperties ?: return
        var aboutText = SLMessage.stringFromVariableUTF(properties.PropertiesData_Field.AboutText)
        var flAboutText = SLMessage.stringFromVariableOEM(properties.PropertiesData_Field.FLAboutText)
        if (isFirstLife()) {
            flAboutText = text
        } else {
            aboutText = text
        }
        agentCircuit.modules.userProfiles.UpdateAvatarProperties(
            properties.PropertiesData_Field.ImageID,
            properties.PropertiesData_Field.FLImageID,
            aboutText,
            flAboutText,
            (properties.PropertiesData_Field.Flags and 1) != 0,
            (properties.PropertiesData_Field.Flags and 2) != 0,
            SLMessage.stringFromVariableOEM(properties.PropertiesData_Field.ProfileURL)
        )
    }

    companion object {
        private const val IS_FIRST_LIFE_KEY = "isFirstLife"

        @JvmStatic
        fun makeSelection(chatterID: ChatterID, isFirstLife: Boolean): Bundle {
            val selection = ChatterFragment.makeSelection(chatterID)
            selection.putBoolean(IS_FIRST_LIFE_KEY, isFirstLife)
            return selection
        }
    }
}
